packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts

```diff
@@ -858,6 +879,18 @@ export async function askApiKeyQuestions(authSettings: $TSObject = undefined) {
   };
 }
 
+export async function askApiKeyExtension() {
+  const apiKeyExtensionQuestion = {
+    message: 'Days to extend API key expiration:',
+    default: '0',
+  };
+  const expirationDaysNum = Number(await prompter.input(
+    apiKeyExtensionQuestion.message,
+    { initial: apiKeyExtensionQuestion.default as string },
+  ));
+  return expirationDaysNum;
+}
```

Comment:
Also, we need that `validate` param to ensure that user does not give us decimals and probably also check for some upper-bound to the number of days key is extended.

Reply:
The upper bound gets a little tricky because it's dependent on the current date as well as the expiration, I tested that an invalid number will return an appropriately displayed message from appsync and I'd rather leave it at that than try to calculate out their service limits. I will put the integer validation in though